drivers/gles3/storage/mesh_storage.cpp

```diff
@@ -255,7 +255,10 @@ void MeshStorage::mesh_add_surface(RID p_mesh, const RS::SurfaceData &p_surface)
 			mesh->bone_aabbs.resize(p_surface.bone_aabbs.size());
 		}
 		for (int i = 0; i < p_surface.bone_aabbs.size(); i++) {
-			mesh->bone_aabbs.write[i].merge_with(p_surface.bone_aabbs[i]);
+			AABB bone = p_surface.bone_aabbs[i];
+			if (!bone.has_no_volume()) {
+				mesh->bone_aabbs.write[i].merge_with(p_surface.bone_aabbs[i]);
+			}
```

Comment:
Sorry to only notice this just after merging, but should it be:
```
const AABB &bone = p_surface.bones_aabbs[i];
```
for a const reference rather than copying the AABB?

And then using `bone` directly in the next line:
```
mesh->bone_aabbs.write[i].merge_with(bone);
```
It is possible the optimizer might optimize this out, but there seems to be little point in copying by value here.

And same in the 2nd section below.

Reply:
> It looks like the name for this optimization is copy elision:

C++17 has guaranteed copy elision. Since Godot uses C++17 in `master`, does this address the issue?

Reply:
I wouldn't rely on anything like that, especially in hot loops. :grin:

Be explicit in what you are asking the compiler to do, and you won't go far wrong. If you are vague, be prepared to examine the assembly. Which will take 10x as long as just writing it correctly in the first place.

Reply:
Ah, I completely forgot about using `bone` on the next line. And yeah, I'll update the code to have the AABB a const ref.